JumpScale9AYS/ays/lib/RunScheduler.py

```
@@ -34,6 +38,18 @@ def __init__(self, repo):
         self._is_running = False
         self._current = None
 
+    def configure_retry_delay(self, retry_config):
+        """
+        sets the retry delays and number
+        @param retry_config: list containing the delay values.
+        """
+        retry_delay = {}
```

why don't you check if the first item of the list contain 0. If it's the case then you know you no retries is required.
Then during iteration of the list, of 0 if encounter, you need to raise an exception.